### PR TITLE
add Coder capabilities doc for code-server users

### PR DESCRIPTION
**Pull request by bpmct**
This is a draft page that code-server users will get recommended if they are running code-server in Kubernetes. Would love feedback on:

- should this live in `/guides` in docs?
- format
- are these the correct capabilities? goal is to highlight Coder features for those doing a DIY code-server + K8s project and are unfamiliar with Coder 
- if/how we should incorporate this with the [comparison](https://coder.com/docs/comparison) doc, which focuses on standalone code-server

### CODE REVIEW

**Comment by khorne3** [comment]
@bpmct I'm a bit confused, so I was hoping you could clarify the messaging; is this doc targeting users who have a DIY code-server setup they're hosting on K8 where we basically say, "Hey, here's what you would get if you used the full Coder deployment?"

**Comment by bpmct** [comment]
> @bpmct I'm a bit confused, so I was hoping you could clarify the messaging; is this doc targeting users who have a DIY code-server setup they're hosting on K8 where we basically say, "Hey, here's what you would get if you used the full Coder deployment?"

Yes exactly, and I'd love feedback on how this could me more clear. A decent number of people attempt to build code-server on Kubernetes without knowledge of Coder's existence. Many of the things highlighted in this page are complex to implement in a DIY solution, but may be necessary for a user.

File: guides/code-server-envs.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+title: Managed code-server Environments
+description: Learn how Coder can improve your code-server deployment on Kubernetes.
+---
+
+If you're deploying [code-server](https://github.com/cdr/code-server) on
+Kubernetes, you may want to consider Coder, our solution for developers and
+development teams. Coder is deployed to Kubernetes
```
Comment (ashumz) [inline guides/code-server-envs.md:8]:
```suggestion
development teams. Coder runs on Kubernetes
```

File: guides/code-server-envs.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+clusters and offers the following features in addition to the base
```
Comment (ashumz) [inline guides/code-server-envs.md:9]:
```suggestion
and offers the following features in addition to the base
```

**Review by kylecarbs — APPROVED** [review]
Needs to be added to the manifest, but LGTM!

**Comment by khorne3** [comment]
Latest changes:

- Moved to Guides/Deployments
- Simplified URL
- Added doc to manifest.json

**Pull request merged**